src/main/java/de/qabel/desktop/hockeyapp/HockeyAppClient.java

```diff
@@ -163,12 +160,19 @@ public HockeyAppVersion createNewVersion(String version) throws IOException {
         request.setEntity(new UrlEncodedFormEntity(parameters, HTTP.UTF_8));
 
         HttpResponse response = httpClient.execute(request);
-        String responseContent = EntityUtils.toString(response.getEntity());
-        JSONObject parsedJson = new JSONObject(responseContent);
 
-        setCurrentHockeyVersion(new HockeyAppVersion(parsedJson.getInt("id"),parsedJson.getString("shortversion")));
+        String responseContent = EntityUtils.toString(response.getEntity());
 
-        return currentHockeyVersion;
+        try {
+            JSONObject parsedJson = new JSONObject(responseContent);
+            int versionId = parsedJson.getInt("id");
+            String shortVersion = parsedJson.getString("shortversion");
+            setCurrentHockeyVersion(new HockeyAppVersion(versionId, shortVersion));
+
+            return currentHockeyVersion;
+        } catch (JSONException e){
+            throw new IOException("Unexpected JSON format return from HokkeyApp", e);
```

Comment:
When you introduce new error handling, first create a test that provokes the error you want to handle. In this case: Create a test that gives the method invalid JSON which produces an unhandled exception. The Tests expects an IOException with a message of "Unexpected JSON format...".

Then, after you made a test that fails, introduce the error handler (the catch JSONException and throw IOException) and your tests are green again.